pkg/connector/aws_account.go

```diff
@@ -340,22 +345,46 @@ func (o *accountResourceType) Grants(
 	}
 }
 
-func accountGrant(resource *v2.Resource, samlRole string, oktaUserId string) *v2.Grant {
+func (o *accountResourceType) accountGrant(resource *v2.Resource, samlRole string, oktaUserId string) *v2.Grant {
+	grantOpts := make([]sdkGrant.GrantOption, 0)
+	if o.connector.awsConfig.AWSSourceIdentityMode {
+		grantOpts = append(grantOpts, sdkGrant.WithAnnotation(&v2.ExternalResourceMatchID{Id: oktaUserId}))
+	}
 	ur := &v2.Resource{Id: &v2.ResourceId{ResourceType: resourceTypeUser.Id, Resource: oktaUserId}}
-	return sdkGrant.NewGrant(resource, samlRole, ur)
+	return sdkGrant.NewGrant(resource, samlRole, ur, grantOpts...)
 }
 
-func accountGrantGroup(resource *v2.Resource, samlRole string, groupId string) *v2.Grant {
+func (o *accountResourceType) accountGrantGroup(resource *v2.Resource, samlRole string, groupId string) *v2.Grant {
+	grantOpts := make([]sdkGrant.GrantOption, 0)
+	if o.connector.awsConfig.AWSSourceIdentityMode {
+		grantOpts = append(grantOpts, sdkGrant.WithAnnotation(&v2.ExternalResourceMatchID{Id: groupId}))
+	}
 	gr := &v2.Resource{Id: &v2.ResourceId{ResourceType: resourceTypeGroup.Id, Resource: groupId}}
-	return sdkGrant.NewGrant(resource, samlRole, gr)
+	return sdkGrant.NewGrant(resource, samlRole, gr, grantOpts...)
 }
 
-func accountGrantGroupExpandable(resource *v2.Resource, samlRole string, groupId string) *v2.Grant {
-	gr := &v2.Resource{Id: &v2.ResourceId{ResourceType: resourceTypeGroup.Id, Resource: groupId}}
-	return sdkGrant.NewGrant(resource, samlRole, gr, sdkGrant.WithAnnotation(&v2.GrantExpandable{
-		EntitlementIds: []string{fmt.Sprintf("group:%s:member", groupId)},
+func (o *accountResourceType) accountGrantGroupExpandable(resource *v2.Resource, samlRole string, groupId string) (*v2.Grant, error) {
+	rID := &v2.ResourceId{ResourceType: resourceTypeGroup.Id, Resource: groupId}
+	gr := &v2.Resource{Id: rID}
+
+	grantOpts := make([]sdkGrant.GrantOption, 0)
+	expandEntitlementId := fmt.Sprintf("group:%s:member", groupId)
+	if o.connector.awsConfig.AWSSourceIdentityMode {
+		ent := sdkEntitlement.NewAssignmentEntitlement(gr, "member")
+		bidEnt, err := bid.MakeBid(ent)
+		if err != nil {
+			return nil, err
+		}
+		expandEntitlementId = bidEnt
+		grantOpts = append(grantOpts, sdkGrant.WithAnnotation(&v2.ExternalResourceMatchID{Id: groupId}))
+	}
+
+	grantOpts = append(grantOpts, sdkGrant.WithAnnotation(&v2.GrantExpandable{
+		EntitlementIds: []string{expandEntitlementId},
 		Shallow:        true,
 	}))
```

Comment:
Will expand grants to the external source?

Reply:
yes, for external resources it will expand, but it relies on bid (since sometimes the external resource match is by attribute/match all and we don't always have the static id) 

https://github.com/ConductorOne/baton-sdk/pull/349